app/Http/Controllers/RestaurantController.php

```diff
@@ -4,16 +4,62 @@
 
 use App\Models\Restaurant;
 use App\Models\Review;
+use App\Models\RestaurantPhoto;
+use App\Models\FoodType;
+use App\Models\AreaType;
+use App\Models\Feature;
+use App\Models\FeatureType;
 use Illuminate\Http\Request;
 
 class RestaurantController extends Controller
 {
     private $restaurant;
     private $review;
+    private $restaurantphoto;
+    private $foodtype;
+    private $areatype;
+    private $feature;
+    private $featuretype;
 
-    public function __construct(Restaurant $restaurant, Review $review){
+    public function __construct(Restaurant $restaurant, Review $review, RestaurantPhoto $restaurantphoto, FoodType $foodtype, AreaType $areatype, Feature $feature, FeatureType $featuretype){
         $this -> restaurant = $restaurant;
         $this -> review = $review;
+        $this -> restaurantphoto = $restaurantphoto;
+        $this -> foodtype = $foodtype;
+        $this -> areatype = $areatype;
+        $this -> feature = $feature;
+        $this -> featuretype = $featuretype;
+    }
+
+    /** Show restaurant ranking page */
+    public function restaurantRanking(){
+        return view('restaurant.ranking');
+    }
+
+    /** Show restaurant detail page */
+    public function ShowRestaurantDetail($id){
+        $restaurant = $this->restaurant->findOrFail($id);
+        $restaurantphoto = $restaurant->restaurant_photos;
+                         //↑restaurant Model.    //↑function on restaurant Model.
+        $foodtype = $this->foodtype->findOrFail($restaurant->foodtype->id);
+        $areatype = $this->areatype->findOrFail($restaurant->areatype->id);
+
+        // featuretype
+        $feature = $restaurant->features;
+        // dd($feature);
+
+
+
+        // $feature = $this->restaurant->findOrFail($restaurant->id);
+        // $all_features = $this->feature->all();
+        // foreach ($all_features as $featuretype) {
+        //     $featuretype[] = $feature->featuretype;
+        // }
+        // $all_featuretypes = $this->featuretype->all();
+
+        return view('restaurant.detail')->with('restaurant', $restaurant)->with('restaurantphoto', $restaurantphoto)->with('foodtype', $foodtype)->with('areatype', $areatype);
```

Comment:
I suggest to use this method instead of nested ->with. It's better to use this one as it has more readability than above.

```
return view('restaurant.detail',
[
    'restaurant' => $restaurant,
    'restaurantphoto' => $restaurantphoto,
    'foodtype' => $foodtype,
    'areatype' => $areatype
]);
```

Reply:
Thank you, I edited!